src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,96 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Post } from '../types/Post';
+import { CommentData } from '../types/Comment';
+
+type Props = {
+  selectedPost: Post;
+  addComment: (comment: CommentData) => Promise<void>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  addComment,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [name, setName] = useState('');
+  const [nameError, setNameError] = useState('');
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [text, setText] = useState('');
+  const [textError, setTextError] = useState('');
```

Comment:
better to make one state input all input fields and one state for errors.
```
const [inputs, setInputs] = useState({
   name: '',
   email: '',
   body: ''
})
```
same for errors.
and use event.target.name for change proper field.

instead three handlers will be one change handler